bee-network/bee-autopeering/src/peer/stores/rocksdb.rs

```diff
@@ -0,0 +1,145 @@
+        let cf = self.open_cf(ACTIVE_PEERS_CF);
+        let key = *active_peer.peer_id();
+
+        self.db.put_cf(&cf, key, active_peer.to_bytes())
+    }
+
+    fn store_all_active(&self, active_peers: &ActivePeersList) -> Result<(), Self::Error> {
+        let cf = self.open_cf(ACTIVE_PEERS_CF);
+
+        let mut batch = WriteBatch::default();
+        active_peers
+            .read()
+            .iter()
+            .for_each(|p| batch.put_cf(&cf, p.peer_id(), p.to_bytes()));
+
+        self.db.write(batch)
+    }
+
+    fn store_replacement(&self, peer: Peer) -> Result<(), Self::Error> {
+        let cf = self.open_cf(REPLACEMENTS_CF);
+        let key = *peer.peer_id();
+
+        self.db.put_cf(&cf, key, peer.to_bytes())
+    }
+
+    fn store_all_replacements(&self, replacements: &ReplacementPeersList) -> Result<(), Self::Error> {
+        let cf = self.open_cf(REPLACEMENTS_CF);
+
+        let mut batch = WriteBatch::default();
+        replacements
+            .read()
+            .iter()
+            .for_each(|p| batch.put_cf(&cf, p.peer_id(), p.to_bytes()));
+
+        self.db.write(batch)
+    }
+
+    fn contains(&self, peer_id: &PeerId) -> Result<bool, Self::Error> {
+        let cf = self.open_cf(ACTIVE_PEERS_CF);
+        if self.db.get_cf(&cf, peer_id)?.is_some() {
+            Ok(true)
+        } else {
+            let cf = self.open_cf(REPLACEMENTS_CF);
+            Ok(self.db.get_cf(&cf, peer_id)?.is_some())
+        }
+    }
+
+    fn fetch_active(&self, peer_id: &PeerId) -> Result<Option<ActivePeer>, Self::Error> {
+        let cf = self.open_cf(ACTIVE_PEERS_CF);
+
+        Ok(self.db.get_cf(&cf, peer_id)?.map(|b| ActivePeer::from_bytes(&b)))
+    }
+
+    fn fetch_all_active(&self) -> Result<Vec<ActivePeer>, Self::Error> {
+        let cf = self.open_cf(ACTIVE_PEERS_CF);
+
+        Ok(self
+            .db
+            .iterator_cf(&cf, IteratorMode::Start)
+            .map(|(_, b)| ActivePeer::from_bytes(&b))
+            .collect::<Vec<_>>())
+    }
+
+    fn fetch_replacement(&self, peer_id: &PeerId) -> Result<Option<Peer>, Self::Error> {
+        let cf = self.open_cf(REPLACEMENTS_CF);
+
+        Ok(self.db.get_cf(&cf, peer_id)?.map(Peer::from_bytes))
+    }
+
+    fn fetch_all_replacements(&self) -> Result<Vec<Peer>, Self::Error> {
+        let cf = self.open_cf(REPLACEMENTS_CF);
+
+        Ok(self
+            .db
+            .iterator_cf(&cf, IteratorMode::Start)
+            .map(|(_, bytes)| Peer::from_bytes(bytes))
+            .collect::<Vec<_>>())
+    }
+
+    fn delete(&self, _: &PeerId) -> Result<bool, Self::Error> {
+        unimplemented!("no need for single entry removal at the moment")
+    }
+
+    fn delete_all(&self) -> Result<(), Self::Error> {
+        self.db.drop_cf(ACTIVE_PEERS_CF)?;
+        self.db.create_cf(ACTIVE_PEERS_CF, &self.config.options)?;
+
+        self.db.drop_cf(REPLACEMENTS_CF)?;
+        self.db.create_cf(REPLACEMENTS_CF, &self.config.options)?;
```

Comment:
I feel like I remember there was an issue proceeding that way, and we had to come up with `truncate` that uses `delete_range_cf`